projects/plugins/jetpack/_inc/lib/core-api/wpcom-endpoints/class-wpcom-rest-api-v2-endpoint-admin-menu.php

```diff
@@ -83,10 +83,6 @@ public function get_item_permissions_check( $request ) { // phpcs:ignore Generic
 	 * @return WP_REST_Response|WP_Error Response object on success, or WP_Error object on failure.
 	 */
 	public function get_item( $request ) { // phpcs:ignore Generic.CodeAnalysis.UnusedFunctionParameter, VariableAnalysis.CodeAnalysis.VariableAnalysis.UnusedVariable
-		if ( ! function_exists( 'wpcom_is_nav_redesign_enabled' ) || ! wpcom_is_nav_redesign_enabled() ) {
```

Comment:
I believe this is redundant. This file is already loaded for self-hosted and WoA [here](https://github.com/Automattic/jetpack/blob/trunk/projects/packages/masterbar/src/class-main.php#L53) and for Simple sites in fbhepr%2Skers%2Sjcpbz%2Sjc%2Qpbagrag%2Szh%2Qcyhtvaf%2Sznfgreone.cuc%3Se%3Qns1125q9%2370-og

Reply:
Yup, I was initially concerned about this change, but double checked that does not cause any regression.

Reply:
Actually I double checked the endpoint response for a self-hosted site thanks to @coder-karen 's comment in Slack p1717600397602839/1717592262.108989-slack-C05PV073SG3 and it looks like there's a difference between the endpoint logic and `Main`:

In the endpoint, we used to check only `if ( ! function_exists( 'wpcom_is_nav_redesign_enabled' ) || ! wpcom_is_nav_redesign_enabled() ) {`

however in `Main` [here](https://github.com/Automattic/jetpack/blob/trunk/projects/packages/masterbar/src/class-main.php#L53) we also take the `jetpack_load_admin_menu_class` filter under consideration:

```
/**
 * Whether to load the admin menu functionality.
 *
 * @use add_filter( 'jetpack_load_admin_menu_class', '__return_true' );
 *
 * @param bool $load_admin_menu_class Load Jetpack's custom admin menu functionality. Default to false.
 */
if ( ! $should_use_nav_redesign && apply_filters( 'jetpack_load_admin_menu_class', false ) ) {
	require_once __DIR__ . '/admin-menu/load.php';
}
```

If we want to be consistent, the removing this check from the endpoint is fine. If however, we don't want to take the `jetpack_load_admin_menu_class` filter into account (existing endpoint behaviour) then I should probably bring this back.

cc @fushar who introduced the conditional in https://github.com/Automattic/jetpack/pull/36601. I noticed that initially you were [indeed using the filter only](https://github.com/Automattic/jetpack/pull/36601/commits/21f2823d5cba3e95c33f69a510cea10f44329276) but then switched to `if ( ! function_exists( 'wpcom_is_nav_redesign_enabled' ) || ! wpcom_is_nav_redesign_enabled() ) {`.

Reply:
Good catch @fgiannar. Turns out the Jetpack beta plugin wasn't targeting this branch on my self-hosted site so I wasn't really testing these changes. After fixing that, I confirm there is an issue in there:

Before | After
--- | ---
<img width="316" alt="Screenshot 2024-06-06 at 10 30 54" src="https://github.com/Automattic/jetpack/assets/1233880/78febcb1-f6a7-47d8-8234-4c4f34e5099c"> | <img width="465" alt="Screenshot 2024-06-06 at 10 30 11" src="https://github.com/Automattic/jetpack/assets/1233880/4c52b2b1-61d0-4e13-bc32-c377bf801337">

Let's see what @fushar (or anyone else from @Automattic/lego) suggests here.

Reply:
Hi all, I am looking at this right now and trying to recall what I did here. 😄

Reply:
Actually the issue is exactly what @mmtr said above. For Jetpack self-hosted sites, when we're in Calypso, we still want to see the nav-unification (left screenshot), whereas after this PR, we're showing default menus, which are incompatible.

> I noticed that initially you were indeed using the filter only but then switched to if ( ! function_exists( 'wpcom_is_nav_redesign_enabled' ) || ! wpcom_is_nav_redesign_enabled() ) {.

It's because I realized the first condition broke Jetpack self-hosted sites for the above reason. (I commented here https://github.com/Automattic/jetpack/pull/36601#issuecomment-2024207809)

Reply:
What happens if the bring back `require_once JETPACK__PLUGIN_DIR . '/modules/masterbar/admin-menu/load.php';` without the check?

Reply:
That would essentially revert https://github.com/Automattic/jetpack/pull/36601; which means we will get tangled menus in Calypso for untangled sites. (Haven't actually tested but I believe this is the case.)

Reply:
Oh I see. We need to restore what we had before then.

Although a better implementation might be to add a `wpcom_is_nav_redesign_enabled` check inside the `should_customize_nav` function, so we can always load the `admin-menu/load.php` file.

Reply:
Awesome, thank you. Seems to be working well now (tested Simple Default, Simple Classic, Atomic Default, Atomic Classic, ad Self-hosted)